packages/legacy/core/App/navigators/MainStack.tsx

```diff
@@ -40,76 +46,84 @@ const MainStack: React.FC = () => {
       opacity: current.progress,
     },
   })
+  useMemo(() => {
+    setHideElements(currentStep === undefined ? 'auto' : 'no-hide-descendants')
+  }, [currentStep])
 
   return (
-    <Stack.Navigator initialRouteName={Screens.Splash} screenOptions={{ ...defaultStackOptions, headerShown: false }}>
-      <Stack.Screen name={Screens.Splash} component={splash} />
-      <Stack.Screen name={Stacks.TabStack} component={TabStack} />
-      <Stack.Screen
-        name={Screens.CredentialDetails}
-        component={CredentialDetails}
-        options={{
-          title: t('Screens.CredentialDetails'),
-          headerShown: true,
-          ...ScreenOptionsDictionary[Screens.CredentialDetails],
-        }}
-      />
-      <Stack.Screen
-        name={Screens.OpenIDCredentialDetails}
-        component={OpenIDCredentialDetails}
-        options={{
-          title: t('Screens.CredentialDetails'),
-          ...ScreenOptionsDictionary[Screens.OpenIDCredentialDetails],
-        }}
-      />
-      <Stack.Screen
-        name={Screens.Chat}
-        component={Chat}
-        options={({ navigation }) => ({
-          headerShown: true,
-          title: t('Screens.CredentialOffer'),
-          headerLeft: () => (
-            <IconButton
-              buttonLocation={ButtonLocation.Left}
-              accessibilityLabel={t('Global.Back')}
-              testID={testIdWithKey('BackButton')}
-              onPress={() => {
-                navigation.navigate(TabStacks.HomeStack, { screen: Screens.Home })
-              }}
-              icon="arrow-left"
-            />
-          ),
-        })}
-      />
-      <Stack.Screen name={Stacks.ConnectStack} component={ConnectStack} />
-      <Stack.Screen
-        name={Stacks.SettingStack}
-        component={SettingStack}
-        options={{
-          cardStyleInterpolator: forFade,
-        }}
-      />
-      <Stack.Screen name={Stacks.ContactStack} component={ContactStack} />
-      <Stack.Screen name={Stacks.NotificationStack} component={NotificationStack} />
-      <Stack.Screen
-        name={Stacks.ConnectionStack}
-        component={DeliveryStack}
-        options={{
-          gestureEnabled: false,
-          cardStyleInterpolator: CardStyleInterpolators.forVerticalIOS,
-          presentation: 'modal',
-        }}
-      />
-      <Stack.Screen name={Stacks.ProofRequestsStack} component={ProofRequestStack} />
-      <Stack.Screen
-        name={Stacks.HistoryStack}
-        component={HistoryStack}
-        options={{
-          cardStyleInterpolator: forFade,
-        }}
-      />
-      {CustomNavStack1 ? <Stack.Screen name={Stacks.CustomNavStack1} component={CustomNavStack1} /> : null}
-    </Stack.Navigator>
+    <SafeAreaView
```

Comment:
I think should just be a View, no? So that the individual screens can apply their own SafeAreaView configs. No bg color needed